test/propagated_params_test.go

```diff
@@ -0,0 +1,161 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/test/parse"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	knativetest "knative.dev/pkg/test"
+)
+
+func TestPropagatedParams(t *testing.T) {
+	t.Parallel()
+	type tests struct {
+		name            string
+		pipelineName    string
+		taskName        string
+		pipelineRunFunc func(*testing.T, string, string, string) *v1beta1.PipelineRun
+	}
+
+	tds := []tests{{
+		name:            "propagated parameters fully",
+		pipelineName:    "propagated-parameters-fully",
+		taskName:        "echo-hello",
+		pipelineRunFunc: getPropagatedParamPipelineRun,
+	}, {
+		name:            "propagated parameters with inner scope",
+		pipelineName:    "propagated-parameters-inner-scope",
+		taskName:        "echo-hello",
+		pipelineRunFunc: getPropagatedParamWithScopePipelineRun,
+	}, {
+		name:            "propagated parameters with default inner scope",
+		pipelineName:    "propagated-parameters-default-inner-scope",
+		taskName:        "echo-hello",
+		pipelineRunFunc: getPropagatedParamWithDefaultScopePipelineRun,
+	}}
+
+	for _, td := range tds {
+		td := td // capture range variable
```

Comment:
sorry can you elaborate what this is for?

Reply:
Honestly, it isnt required. I was probably required in the test that I copied and edited. I cleaned it up to avoid confusion.

Reply:
I think we do need that actually - see https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables and https://internals.tidb.io/t/topic/391

Reply:
Oh I see. And because we're running the tests in parallel, we need it.